apps/api/views/chat.py

```diff
@@ -55,13 +62,63 @@ def check_experiment_access(experiment, participant_id):
     return None
 
 
-def check_session_access(session):
+def handle_embedded_widget_auth(request, experiment_id=None, session=None):
+    embed_key = request.headers.get("X-Embed-Key")
+    if not embed_key:
+        return False, None, None
+
+    # Extract origin domain from headers
+    origin_domain = extract_domain_from_headers(request)
+    if not origin_domain:
+        raise PermissionDenied("Origin or Referer header required for embedded widgets")
+
+    if experiment_id:
+        is_valid, experiment_channel = validate_embed_key_for_experiment(
+            token=embed_key, origin_domain=origin_domain, experiment_id=experiment_id
+        )
+    elif session:
+        is_valid, experiment_channel = validate_embedded_widget_request(
```

Comment:
Can this not also use the `validate_embed_key_for_experiment` method? We have the experiment on the session. This way, the `ExperimentChannel` query can be scoped to the correct experiment and if the channel exists, we don't have to do the `session.experiment_channel.platform != ChannelPlatform.EMBEDDED_WIDGET` check as well on the following line

Reply:
+1